specs/agents/tracing-instrumentation-azure-functions.md

```diff
@@ -0,0 +1,111 @@
+actual `faas.trigger.type`. For trigger types that are specifically supported (e.g HTTP) these values
+can vary on trigger-specific specifications apply.
+
+| Field | Value | Description | Source |
+| - | - | - | - |
+| `name` | e.g. `MySampleTrigger` | The transaction name. Use function name if trigger type is `other`. | *generic context* |
+| `type` | e.g. `request`, `messaging` | The transaction type. | Use `request` if trigger type is undefined. |
+| `outcome` | `success`/`failure` | Set to `failure` if a function error can be detected otherwise `success`. | |
+| `result` | `success`/`failure` |Set to `failure` if a function error can be detected, otherwise `success`. | Trigger specific. |
+| `faas.name` | e.g. `MyFunctionApp/MySampleTrigger` | The function app name and the function name, using this format: `<FUNCTION_APP_NAME>/FUNCTION_NAME>`. | *generic context* |
+| `faas.id` | e.g. `/subscriptions/d2ba53be-0815-4...` | The [fully qualified resource ID](https://learn.microsoft.com/en-us/rest/api/resources/resources/get-by-id) of the Azure Function, which has this format: `/subscriptions/<SUBSCRIPTION_GUID>/resourceGroups/<RG>/providers/Microsoft.Web/sites/<FUNCAPP>/functions/<FUNC>` | *generic context*, environment |
+| `faas.trigger.type` | `other` | The trigger type. Use `other` if the trigger type is unknown or cannot be specified. | More concrete triggers are `http`, `pubsub`, `datasource`, `timer` (see specific triggers below). |
+| `faas.execution` | `203621a2-62f...` | The unique invocation id of the function. | *generic context* |
+| `faas.coldstart` | `true`/`false` | A boolean value indicating whether this function invocation was a cold start or not. See the [Deriving cold starts](#deriving-cold-starts) section below. |
+
+### Metadata
+
+Generic cloud metadata fetching SHOULD be disabled when the agent is running in an Azure Functions app. It will not
+yield any useful data and only incur additional cold start overhead due to the non-existent HTTP metadata endpoints.
+
+Agents SHOULD rather:
+
+* Check for the existence of the `FUNCTIONS_WORKER_RUNTIME` environment variable that indicates that the agent is running
+in an Azure Function application.
+* Perform specific cloud metadata detection for Azure Functions (see [Metadata](./metadata.md)).
+
+In addition to those described in [Metadata](./metadata.md), following metadata fields are relevant for Azure Functions:
+
+| Field | Value | Description | Source |
+| - | - | - | - |
+| `service.framework.name` | `Azure Functions` | Constant value for the framework name. | |
+| `service.framework.version` | e.g. `~4` | Version of the Azure Functions runtime. | `FUNCTIONS_EXTENSION_VERSION` |
+| `service.runtime.name`| e.g. `dotnet-isolated` | The language worker runtime (see [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#functions_worker_runtime)). | `FUNCTIONS_WORKER_RUNTIME` |
+
```

Comment:
```suggestion
| `service.node.configured_name` | e.g. '25d4009bce1d ...' | Unique ID of the VM instance. | `WEBSITE_INSTANCE_ID` ([Azure docs](https://learn.microsoft.com/en-us/azure/app-service/reference-app-settings#scaling)) |
```

The `service.node.configured_name` is preferred over `system.hostname` for the entries in the "Instances" table on the service overview page, e.g.:
<img width="1065" alt="Screen Shot 2022-12-15 at 2 15 25 PM" src="https://user-images.githubusercontent.com/46866/207978818-f5d50633-5a2c-41a3-bbc1-565c82bd944c.png">
(The '8c1ff...' is from `WEBSITE_INSTANCE_ID`; the "10-20-0-211" is the hostname already sent in metadata at `system.hostname`.)

Reply:
For Lambda, we disabled the instances view in the UI. Should we do the same for Azure functions?

Reply:
@AlexanderWert Oh interesting. IIUC we have "Active Instances" on the Metrics page for Lambda (per https://github.com/elastic/kibana/issues/142328). If we had an equivalent for Azure functions, then yes, the "Instances" view would be somewhat redundant.

Azure Functions docs feel less concrete than AWS Lambda docs on the lifetime of the instances. The Azure Function VMs are *re-used* for deployment updates of function apps, so they are somewhat less ephemeral than Lambda instances.  Also, given that a single Azure Function instance can be concurrently handling requests for separate Functions in the same Function App, and hence they can have impact on each other (share CPU/memory), there is perhaps a *little* more of a basis for showing possible correlation of perf issues.

Reply:
As discussed yesterday @trentm, let's add the property to the spec.
We can then decide later whether we want to leverage that data in the UI.